src/Admin/SiteHealth.php

```diff
@@ -497,4 +519,60 @@ public function add_extensions( $extensions ) {
 			]
 		);
 	}
+
+	/**
+	 * Re-enable the CSS Transient caching.
+	 *
+	 * This is triggered via an AJAX call from the Site Health panel.
+	 */
+	public function reenable_css_transient_caching() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_send_json_error( 'Unauthorized.', 401 );
+		}
+
+		$result = AMP_Options_Manager::update_option( Option::DISABLE_CSS_TRANSIENT_CACHING, false );
+
+		if ( false === $result ) {
+			wp_send_json_error( 'CSS transient caching could not be re-enabled.', 500 );
+		}
+
+		wp_send_json_success( 'CSS transient caching was re-enabled.', 200 );
+	}
+
+	/**
+	 * Add needed styles for the Site Health integration.
+	 */
+	public function add_styles() {
+		echo '
+            <style>
+                .wp-core-ui .button.reenable-css-transient-caching ~ .success-icon,
+                .wp-core-ui .button.reenable-css-transient-caching ~ .success-text,
+                .wp-core-ui .button.reenable-css-transient-caching ~ .failure-icon,
+                .wp-core-ui .button.reenable-css-transient-caching ~ .failure-text {
+                    display: none;
+                }
+
+                .wp-core-ui .button.reenable-css-transient-caching ~ .success-icon,
+                .wp-core-ui .button.reenable-css-transient-caching ~ .failure-icon {
+                    font-size: xx-large;
+                    padding-right: 1rem;
+                }
+
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-success ~ .success-icon,
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-success ~ .success-text,
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-failure ~ .failure-icon,
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-failure ~ .failure-text {
+                    display: inline-block;
+                }
+
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-success ~ .success-icon {
+                    color: #46b450;
+                }
+
+                .wp-core-ui .button.reenable-css-transient-caching.ajax-failure ~ .failure-icon {
+                    color: #dc3232;
+                }
+            </style>
+        ';
```

Comment:
It seems spaces are being used here when tabs are expected.

Reply:
Wouldn't it be better to have spaces in the HTML markup rather than tabs?

Also, things like this would be nicer via views so you can split the logic from the markup. We should discuss at one point whether we want to have a very simple view abstraction (can be a single class if we want to keep it simple) to pull in external (templated) files.

Reply:
Coding standard for HTML is to use tabs: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/#indentation

Reply:
Love the code examples on that page, btw... :)